src/main/java/org/springframework/retry/backoff/ExponentialBackOffPolicy.java

```diff
@@ -289,15 +289,25 @@ public synchronized long getSleepAndIncrement() {
 		}
 
 		protected long getNextInterval() {
-			return (long) (this.interval * getMultiplier());
+			if (this.initialIntervalSupplier == null || this.interval != DEFAULT_INITIAL_INTERVAL) {
+				return (long) (this.interval * getMultiplier());
+			}
+			else {
+				return (long) (this.initialIntervalSupplier.get() * getMultiplier());
```

Comment:
This is incorrect - the multiplier must be applied to the previous interval; this always applies it to the initial interval.

Reply:
Yes, that's why there is that logic in the `if`: `|| this.interval != DEFAULT_INITIAL_INTERVAL`.
So, the `initialIntervalSupplier` is called only once, for the first time. The next `interval` is really based on the previous one from now on.

Reply:
I've made this by purpose: during first iteration we should check if supplier provided or not, if provided first iteration should use value provided by `initialIntervalSupplier`(if present), further it will use `interval` value since it will be assigned to `interval` [here ](https://github.com/spring-projects/spring-retry/blob/27011f1bc787e3af9caca1369a5bc02a4c7c57e6/src/main/java/org/springframework/retry/backoff/ExponentialBackOffPolicy.java#L286)

Reply:
But, if the supplier happens to return `DEFAULT_INITIAL_INTERVAL`, it will be called every time.